src/ui/training/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as xhr from './xhr'
+import { PuzzleSyncData, PuzzleData } from '../../lichess/interfaces/training'
+import router from '../../router'
+import settings from '../../settings'
+import { OfflinePuzzle } from './interfaces'
+
+export function syncPuzzles() {
+  const unsolved = settings.training.unsolvedPuzzles
+  const user = settings.training.user()
+  const curRating = user ? user.rating : null
+  const discardedPuzzles: OfflinePuzzle[] = []
+  // Cull puzzles that were retreived when the user's rating was significantly different than it is now
+  if (curRating)
+    unsolved(unsolved().reduce((acc: OfflinePuzzle[], cur: OfflinePuzzle) => {
+      const ratingDiff = (curRating && cur.userRating) ? Math.abs(curRating - cur.userRating) : 0
+      if (ratingDiff < settings.training.ratingDiffThreshold) {
+        discardedPuzzles.push(cur)
+        acc.push(cur)
```

Comment:
You should not mutate `acc` directly. Instead, return new `acc` like: `return acc.concat([cur])`